sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPlugin.cs

```diff
@@ -39,7 +39,7 @@ internal abstract class ServiceBusPlugin
 #pragma warning disable CA1822 // Mark members as static
         protected void SetBody(ServiceBusReceivedMessage message, BinaryData body)
         {
-            message.AmqpMessage.Body = new AmqpDataMessageBody(new ReadOnlyMemory<byte>[] { body });
+            message.AmqpMessage.Body = new AmqpMessageBody(new ReadOnlyMemory<byte>[] { body });
```

Comment:
Seems like the case of "pass a single `BinaryData` in for the sequence is common;  should we have a constructor overload that implicitly creates the array and lets callers pass a single body?

Reply:
This is still internal for now, so we can look at the design when we enable plugins.